x/nexus/keeper/rate_limit.go

```diff
@@ -0,0 +1,130 @@
+package keeper
+
+import (
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/axelar-core/utils"
+	"github.com/axelarnetwork/axelar-core/utils/events"
+	"github.com/axelarnetwork/axelar-core/utils/key"
+	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	"github.com/axelarnetwork/axelar-core/x/nexus/types"
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+// RateLimitTransfer applies a rate limit to transfers, and returns an error if the rate limit is exceeded
+func (k Keeper) RateLimitTransfer(ctx sdk.Context, chain exported.ChainName, asset sdk.Coin, outgoing bool) error {
+	rateLimit, found := k.getRateLimit(ctx, chain, asset.Denom)
+	if !found {
+		return nil
+	}
+
+	epoch := uint64(ctx.BlockTime().UnixNano() / rateLimit.Window.Nanoseconds())
+
+	transferRate, found := k.getTransferRate(ctx, chain, asset.Denom, outgoing)
+	if !found || transferRate.Epoch != epoch {
```

Comment:
After discussing with the team during standup, we figure it's better to just delete the existing `transferRate` whenever a new transfer limit is set.

Reply:
I don't see this addressed yet? @milapsheth

Reply:
it's being deleted in `SetRateLimit`